### PR TITLE
Cable charger bionic

### PR DESCRIPTION
**Pull request by Coolthulhu**
Bionic that eats up vehicle power through a held jumper cable (doesn't need to be wielded, but needs to be in inventory).
Eats up 10 units of vehicle power to regenerate 1 point of bionic power per turn.
Will try all the active cables until it gets that 10 vehicle power.

Limitations:
- Requires a jumper cable to use
- Can't use excess from vehicle power generation, power must be generated and deposited into battery first

### CODE REVIEW

**Comment by Mecares** [comment]
Does it allow to transfer in both directions?

File: src/bionics.cpp [hunk 1]
```diff
@@ -595,6 +595,14 @@ bool player::activate_bionic( int b, bool eff_only )
             tank_plut = 0;
             reactor_plut = 0;
             }
+    } else if (bio.id == "bio_cable") {
+        bool has_cable = has_item_with( []( const item &it ) {
+            return it.has_flag( "CABLE_SPOOL" ) && it.active;
```
Comment (codemime) [inline src/bionics.cpp]: `it.active` is much cheaper. I'd put it before `it.has_flag( "CABLE_SPOOL" )` (but it's not a big deal).

File: src/bionics.cpp [hunk 2]
```diff
@@ -723,6 +731,33 @@ void player::process_bionic( int b )
     } else if (bio.id == "bio_hydraulics") {
         // Sound of hissing hydraulic muscle! (not quite as loud as a car horn)
         sounds::sound( pos(), 19, _("HISISSS!"));
+    } else if( bio.id == "bio_cable" ) {
+        const std::vector<item*> cables = items_with( []( const item &it ) {
+            return it.active && it.has_flag( "CABLE_SPOOL" );
```
Comment (codemime) [inline src/bionics.cpp]: Some minor duplication here. Please consider assigning the lambda to a constant.

File: src/bionics.cpp [hunk 3]
```diff
@@ -723,6 +731,33 @@ void player::process_bionic( int b )
...
+        int wants_power_amt = battery_per_power;
+        for( const item *cable : cables ) {
+            const auto &target = cable->get_cable_target();
+            vehicle *veh = g->m.veh_at( target );
+            if( veh == nullptr ) {
+                continue;
+            }
+
+            int got_power = wants_power_amt - veh->discharge_battery( wants_power_amt );
+            wants_power_amt -= got_power;
```
Comment (codemime) [inline src/bionics.cpp]: If I get it right:
`wants_power_amt = wants_power_amt - wants_power_amt + veh->discharge_battery( wants_power_amt );`
or:
`wants_power_amt = veh->discharge_battery( wants_power_amt );`
Was that really intended?

Comment (Coolthulhu) [inline src/bionics.cpp]:
```
wants_power_amt = veh->discharge_battery( wants_power_amt );
```

Yeah, this one sounds right

**Comment by Coolthulhu** [comment]
Cleaned up

**Pull request merged**